### PR TITLE
Finally finished the ✨ wallet!

### PR DESCRIPTION
**Pull request by bejanaf**
Wallet.js: 
- added component to render liked currencys 
- added WalletOverview.js to integrate the wallet overview with rendered buy and sell orders through a modal
- added AddForm.js to integrate the formular for buy and sell orders through a modal
- added function to calculate the total Balance

All Files:
- added PropTypes
- added Styleguidist
- some little bug fixes

- made some little styling like the WelcomePage.js and header in main pages

### CODE REVIEW

File: client/src/components/CoinsCard.js [hunk 1]
```diff
@@ -12,51 +13,53 @@ export default function Coinscard({ topCoins, onToggleFavoriteCoins }) {
...
+          <CoinNameSymbole>
+            <CoinImage src={topCoin.image} alt="Image of Coin" />
+            <span>
+              {topCoin.name}{' '}
+              <FavoriteIcon
+                src={favoriteIcon}
+                alt="Star image of icon"
+                onClick={() => onToggleFavoriteCoins(topCoin)}
+                style={
+                  topCoin.isFavorite ? { opacity: '100%' } : { opacity: '30%' }
```
Comment (SabMueller) [inline client/src/components/CoinsCard.js:27]: maybe you remove this inline type of styling and move it to the styled components section? :)

File: client/src/components/FavoriteCoins.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          <CoinNameSymbole>
+            <CoinImage src={topCoin.image} alt="Image of Coin" />
+            <span>
+              {topCoin.name}{' '}
+              <FavoriteIcon
+                src={favoriteIcon}
+                alt="Star image of icon"
+                onClick={() => onToggleFavoriteCoins(topCoin)}
+                style={
+                  topCoin.isFavorite ? { opacity: '100%' } : { opacity: '30%' }
```
Comment (SabMueller) [inline client/src/components/FavoriteCoins.js:33]: same as above :)

File: client/src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components';
+import { ReactComponent as LogoIcon } from '../image/HeaderLogo.svg';
+
+export default function ShowLogo() {
```
Comment (SabMueller) [inline client/src/components/Header.js]: consider to name your components and files in the same way :) it makes it easier to track your components!

File: client/src/components/FavoriteCoins.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
+import styled from 'styled-components/macro';
+import PropTypes from 'prop-types';
+import favoriteIcon from '../image/star.png';
+
+export default function FavoriteCoinsCard({
```
Comment (SabMueller) [inline client/src/components/FavoriteCoins.js]: here you could also consider to match the name of the component with the actual file name

File: client/src/components/FavoriteCoins.js (hunk 2 above)
Comment (SabMueller) [inline client/src/components/FavoriteCoins.js]: consider removing the inline styles here as well and move them to the styled components section :)

File: client/src/components/NewsCard.js [hunk 5]
```diff
@@ -1,54 +1,43 @@
...
-                    Go to full article
-                  </a>
-                </p>
-              </ArticleWrapper>
-            )}
-          </section>
-        ))}
+      <ArticleWrapper>
+        <FavoriteIcon
+          src={article.isFavorite ? favoriteNewsIconTwo : favoriteNewsIconOne}
```
Comment (SabMueller) [inline client/src/components/NewsCard.js:11]: nice! i really like this ternary! :)

File: client/src/pages/AddForm.js [hunk 6]
```diff
@@ -0,0 +1,226 @@
+import styled from 'styled-components/macro';
+import PropTypes from 'prop-types';
+import { useState, useEffect } from 'react';
+import { saveToLocal } from '../lib/localStorage';
+import { ReactComponent as Close } from '../image/close.svg';
+import validateEntry from '../lib/validation';
```
Comment (SabMueller) [inline client/src/pages/AddForm.js:5]: consider importing your pages before svgs :)

File: client/src/pages/Wallet.js [hunk 7]
```diff
@@ -1,25 +1,114 @@
-import styled from 'styled-components';
+import styled from 'styled-components/macro';
+import PropTypes from 'prop-types';
 import Navigation from '../components/Navigation';
+import FavoriteCoinsCard from '../components/FavoriteCoins.js';
```
Comment (SabMueller) [inline client/src/pages/Wallet.js]: consider importing your hooks before importing components or other stuff :)

File: client/src/pages/Wallet.js [hunk 8]
```diff
@@ -1,25 +1,114 @@
...
+export default function Wallet({
+  favoriteCoins,
+  exchanges,
+  walletOverview,
+  selectedCoin,
+  onSetWalletOverview,
+  onToggleFavoriteCoins,
+  onSetSelectedCoin,
+}) {
+  const [historicalCoins, setHistoricalCoins] = useState([]);
```
Comment (SabMueller) [inline client/src/pages/Wallet.js]: I dont think you ever use historical Coins throughout your whole app, maybe its best to remove them if you - apparently - dont need them? :)

File: client/src/pages/Wallet.js [hunk 9]
```diff
@@ -1,25 +1,114 @@
...
+
+  useEffect(() => {
+    saveToLocal('portfolioCoins', portfolioCoins);
+  }, [portfolioCoins]);
+
+  function addCoin(portfolioCoin) {
+    setPortfolioCoins([...portfolioCoins, portfolioCoin]);
+  }
+
+  useEffect(() => {
```
Comment (SabMueller) [inline client/src/pages/Wallet.js:34]: nice useEffect! :D

File: client/src/pages/WelcomePage.js [hunk 10]
```diff
@@ -1,43 +1,66 @@
...
+
+  h1 {
+    margin-top: 2rem;
   }
 `;
 
+const Headliner = styled.h1`
+  font-family: neon;
+  font-size: 2rem;
+  color: hsl(330, 100%, 85%);
```
Comment (SabMueller) [inline client/src/pages/WelcomePage.js]: alphabetical order :)

File: client/src/pages/Wallet.js [hunk 11]
```diff
@@ -1,25 +1,114 @@
...
+  onToggleFavoriteCoins,
+  onSetSelectedCoin,
+}) {
+  const [historicalCoins, setHistoricalCoins] = useState([]);
+  const [formView, setFormView] = useState(false);
+  const [portfolioCoins, setPortfolioCoins] = useState(
+    loadFromLocal('portfolioCoins') ?? []
+  );
+  const [totalCoinBalance, setTotalCoinBalance] = useState(0);
+  console.log('Balance:', totalCoinBalance);
```
Comment (SabMueller) [inline client/src/pages/Wallet.js]: you know what to do here ... :)

**Review by SabMueller — APPROVED** [review]
Generally looks good to me! Clean, organized code. Just a few smaller things to consider! ❤️  ✨  Love how your app turned out!

File: client/src/components/Header.js (hunk 3 above)
Comment (bejanaf) [reply client/src/components/Header.js]: fixed :)

File: client/src/components/NewsCard.js (hunk 5 above)
Comment (bejanaf) [reply client/src/components/NewsCard.js:11]: Thank you very much :blush:

File: client/src/pages/AddForm.js (hunk 6 above)
Comment (bejanaf) [reply client/src/pages/AddForm.js:5]: fixed :)

File: client/src/pages/Wallet.js (hunk 7 above)
Comment (bejanaf) [reply client/src/pages/Wallet.js]: fixed :)

File: client/src/pages/Wallet.js (hunk 8 above)
Comment (bejanaf) [reply client/src/pages/Wallet.js]: removed :D

File: client/src/pages/Wallet.js (hunk 11 above)
Comment (bejanaf) [reply client/src/pages/Wallet.js]: Yes, removed that part, thank you :)

File: client/src/pages/Wallet.js (hunk 9 above)
Comment (bejanaf) [reply client/src/pages/Wallet.js:34]: Thank you very much ❤️

**Comment by bejanaf** [comment]
Thank you very much for your code review, that was a big help to improve my code ❤️

**Pull request merged**